packages/ember-cli-eyeglass/src/index.ts

```diff
@@ -77,10 +83,72 @@ function localEyeglassAddons(addon): Array<{path: string}> {
   return paths;
 }
 
+interface EyeglassProjectInfo {
+  apps: Array<any>;
+}
+interface EyeglassAddonInfo {
+  name: string;
+  parentPath: string;
+  isApp: boolean;
+  app: any; // is this safe to cache across builds?
+}
+interface EyeglassAppInfo {
+  assets: BroccoliSymbolicLinker;
+  sessionCache: Map<string, string | number>;
+}
+
 const EMBER_CLI_EYEGLASS = {
   name: 'ember-cli-eyeglass',
+  included(parent) {
+    this._super.included.apply(this, arguments);
+    this.initSelf();
+  },
+  initSelf() {
+    if (EYEGLASS_INFO_PER_ADDON.has(this)) return;
+    let app = findHost(this);
+    if (!app) return;
+    let isApp = (this.app === app);
+    let name = app.name;
+    if (!isApp) {
+      let thisName = typeof this.parent.name === "function" ? this.parent.name() : this.parent.name;
+      name = `${name}/${thisName}`
+    }
+    let parentPath = this.parent.root;
+    if (isApp) {
+      apps.push(app);
+      // we create the symlinker in persistent mode because there's not a good
+      // way yet to recreate the symlinks when sass files are cached. I would
+      // worry about it more but it seems like the dist directory is cumulative
+      // across builds anyway.
+      EYEGLASS_INFO_PER_APP.set(app, {
+        sessionCache: new Map(),
+        assets: new BroccoliSymbolicLinker({}, {annotation: app.name, persistentOutput: true})
```

Comment:
Making `persistentOutput` a parameter of `BroccoliSymbolicLinker` then requires `BroccoliSymbolicLinker`'s `build` method conditionally handling it's own cleanup. Currently it does not, and likely results in links never being removed, only added.

Reply:
Yes, you're correct. To remove those files requires tracking if every url in every addon that added it was removed. If the source changes in a way that would cause the image to no longer be installed, then the reference to that image is also removed. I know that means there's an image still floating around in the output directory until the next restart but it's a ton of work to fix a problem that basically isn't an issue in practical usage because there's not stray urls floating around to the output location in the authored source code.